tests/net/socket/misc/src/main.c

```diff
@@ -51,6 +51,9 @@ ZTEST_USER(socket_misc_test_suite, test_inet_pton)
 
 	res = zsock_inet_pton(AF_INET6, "a:b:c:d:0:1:2:3z", buf);
 	zassert_equal(res, 0, "");
+
+	res = zsock_inet_pton(AF_INET, "192.0.2.400", buf);
+	zassert_equal(res, 0, "");
```

Comment:
Reading the implementation, I think you can easily read beyond the data. If the data is provided from an external source, and for example `"0.0"` is passed, this will call `strtol` to the data after the string (twice).